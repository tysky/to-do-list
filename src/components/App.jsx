import React from 'react';
import Tasks from './TasksList';
import NewTaskInput from './NewTaskInput';

function App() {
  return (
    <>
      <NewTaskInput />
      <Tasks />
    </>
  );
}

export default App;
