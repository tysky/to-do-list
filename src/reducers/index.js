import { combineReducers } from 'redux';
import { handleActions } from 'redux-actions';
import * as actions from '../actions';


const tasksFetchingState = handleActions({
  [actions.initFetchTasksRequest]() {
    return 'requested';
  },
  [actions.initFetchTasksSuccess]() {
    return 'failed';
  },
  [actions.initFetchTasksFailure]() {
    return 'successed';
  },
}, 'none');

const tasks = handleActions({
  [actions.initFetchTasksSuccess](state, { payload }) {
    return payload;
  },
  [actions.newTaskFetched](state, { payload }) {
    return { ...state, [payload.id]: payload };
  },
}, {});

const newTaskInputText = handleActions({
  [actions.addTasksSuccess]() {
    return '';
  },
  [actions.updateNewTaskText](state, { payload: { text } }) {
    return text;
  },
}, '');

export default combineReducers({
  tasks,
  tasksFetchingState,
  newTaskInputText,
});
