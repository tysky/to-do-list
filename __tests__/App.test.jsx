
import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import App from '../src/components/App';
import reducers from '../src/reducers';

const initialState = {
  tasks: {},
  tasksFetchingState: 'none',
  taskSendingState: 'none',
  newTaskInputText: '',
};

const store = createStore(reducers, initialState);

test('App', () => {
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
