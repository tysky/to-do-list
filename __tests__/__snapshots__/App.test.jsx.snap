// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="new-task"
    >
      Add Task
      <input
        autoComplete="off"
        disabled={false}
        id="new-task"
        onChange={[Function]}
        required={true}
        type="text"
        value=""
      />
    </label>
    <button
      disabled={false}
      type="submit"
    >
      Add
    </button>
  </form>,
  <h3>
    Todo
  </h3>,
  <ul />,
]
`;
